### PR TITLE
V2 transactions

### PR DESCRIPTION
**Pull request by bpdsw**

### CODE REVIEW

File: src/main/java/org/jarling/v2/api/FeedItemsResource.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.jarling.exceptions.StarlingBankRequestException;
+import org.jarling.v2.models.transactionfeed.FeedItem;
+import org.jarling.v2.models.transactionfeed.FeedItemAttachment;
+import org.jarling.v2.models.transactionfeed.SpendingCategory;
+
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+public interface FeedItemsResource {
```
Comment (broder) [inline src/main/java/org/jarling/v2/api/FeedItemsResource.java]: `TransactionFeedResource` might be a little more consistent with their API docs.

File: src/main/java/org/jarling/v2/api/FeedItemsResource.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @param accountUid  Account uid (required)
+     * @param categoryUid Category uid (required)
+     * @param feedItemUid Feed item uid (required)
+     * @return FeedItemAttachments
+     * @throws StarlingBankRequestException when there was an issue accessing the resource
+     */
+    List<FeedItemAttachment> getFeedItemAttachments(UUID accountUid, UUID categoryUid, UUID feedItemUid) throws StarlingBankRequestException;
+
+    // TODO download feed item attachment endpoint
```
Comment (broder) [inline src/main/java/org/jarling/v2/api/FeedItemsResource.java]: Did you manage to ask Adam for an example of this?

File: src/main/java/org/jarling/v2/models/transactionfeed/Status.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.jarling.v2.models.transactionfeed;
+
+import lombok.Getter;
+
+@Getter
```
Comment (broder) [inline src/main/java/org/jarling/v2/models/transactionfeed/Status.java]: Nice! Do you want to add `@Getter` to the other v2 enums?

File: src/main/java/org/jarling/v2/Starling.java [hunk 4]
```diff
@@ -109,8 +114,8 @@ public Individual getIndividual() throws StarlingBankRequestException {
     }
 
     @Override
-    public void updateEmail(EmailUpdateRequest emailUpdateRequest) throws StarlingBankRequestException {
-        apiService.put("/account-holder/individual/email", null, null, gson.toJson(emailUpdateRequest));
+    public void updateEmail(String email) throws StarlingBankRequestException {
+        apiService.put("/account-holder/individual/email", null, null, simpleJsonWrapper("email", email));
```
Comment (broder) [inline src/main/java/org/jarling/v2/Starling.java]: What's the rationale behind this over a dedicated model?

File: src/main/java/org/jarling/v2/api/FeedItemsResource.java (hunk 2 above)
Comment (bpdsw) [reply src/main/java/org/jarling/v2/api/FeedItemsResource.java]: I did not - I'll see if I can do that this morning

File: src/main/java/org/jarling/v2/models/transactionfeed/Status.java (hunk 3 above)
Comment (bpdsw) [reply src/main/java/org/jarling/v2/models/transactionfeed/Status.java]: Sure!

File: src/main/java/org/jarling/v2/Starling.java (hunk 4 above)
Comment (bpdsw) [reply src/main/java/org/jarling/v2/Starling.java]: It's friendlier to the consumer of the Jarling library. In this case, simply updating the email with a string is more straightforward than creating a dedicated object that does nothing but wrap a string, and I couldn't see any disadvantages. It was more obvious to me in the response models - if I ask for a list of transactions, I want to receive a `List<Transaction>`, not an object with a single field containing a list of transactions.

Comment (bpdsw) [reply src/main/java/org/jarling/v2/Starling.java]: (The code for the V1 API does the same thing - look for usages of `fromJsonList`

File: src/main/java/org/jarling/v2/api/FeedItemsResource.java (hunk 2 above)
Comment (bpdsw) [reply src/main/java/org/jarling/v2/api/FeedItemsResource.java]: It's now implemented, and successfully tested :)

File: src/main/java/org/jarling/v2/api/TransactionFeedResource.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param accountUid  Account uid (required)
+     * @param categoryUid Category uid (required)
+     * @param feedItemUid Feed item uid (required)
+     * @return FeedItemAttachments
+     * @throws StarlingBankRequestException when there was an issue accessing the resource
+     */
+    List<FeedItemAttachment> getFeedItemAttachments(UUID accountUid, UUID categoryUid, UUID feedItemUid) throws StarlingBankRequestException;
+
+    FeedItemAttachmentData getFeedItemAttachment(UUID accountUid, UUID categoryUid, UUID feedItemUid, UUID feedItemAttachmentUid) throws StarlingBankRequestException;
```
Comment (broder) [inline src/main/java/org/jarling/v2/api/TransactionFeedResource.java]: Should we put some JavaDocs here?

Comment (bpdsw) [reply src/main/java/org/jarling/v2/api/TransactionFeedResource.java]: Well spotted, will fix

**Pull request merged**